src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -42,24 +51,39 @@ public class PipelineCacheServer
                     return;
                 }
 
+                string uploadPath = await this.GetUploadPathAsync(preferredContentFormat, context, path, cancellationToken);
                 //Upload the pipeline artifact.
                 PipelineCacheActionRecord uploadRecord = clientTelemetry.CreateRecord<PipelineCacheActionRecord>((level, uri, type) =>
                     new PipelineCacheActionRecord(level, uri, type, nameof(dedupManifestClient.PublishAsync), context));
                 PublishResult result = await clientTelemetry.MeasureActionAsync(
                     record: uploadRecord,
                     actionAsync: async () =>
                     {
-                        return await dedupManifestClient.PublishAsync(path, cancellationToken);
+                        return await dedupManifestClient.PublishAsync(uploadPath, cancellationToken);
                     });
-
-                CreatePipelineCacheArtifactOptions options = new CreatePipelineCacheArtifactOptions
+        
+                CreatePipelineCacheArtifactContract options = new CreatePipelineCacheArtifactContract
                 {
                     Fingerprint = fingerprint,
                     RootId = result.RootId,
                     ManifestId = result.ManifestId,
                     ProofNodes = result.ProofNodes.ToArray(),
+                    ContentFormat = preferredContentFormat
                 };
 
+                // delete archive file if it's tar.
+                if (string.Equals(preferredContentFormat, ContentFormatConstants.SingleTar, StringComparison.OrdinalIgnoreCase))
+                {
+                    try
+                    {
+                        if (File.Exists(uploadPath))
```

Comment:
this should only happen if the user has a file named "archive.tar" no?  In that case we should bail instead of deleting their file

Reply:
This will only happen if the upload format is tar, so I think we should be good. :)